.circleci/build_docs/commit_docs.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -ex
+
+
+if [ "$1" == "" ]; then
+    echo call as $0 "<target branch>", where branch should be "master" or "1.7" or so
+    exit 1
+fi
+
+set -ex
+echo "committing docs to ${target}"
+
+git checkout gh-pages
+rm -rf docs/$target/*
+cp -r doc/build/html/* docs/$target
+if [ $target == "master" ]; then
+    rm -rf docs/_static/*
+    cp -r doc/build/html/_static/* docs/_static
+fi
+git add docs || true
+git config user.email "soumith+bot@pytorch.org"
+git config user.name "pytorchbot"
+# If there aren't changes, don't make a commit; push is no-op
+git commit -m "auto-generating sphinx docs" || true
+git push -u origin gh-pages
```

Comment:
We should separate out the building and the pushing of docs into two separate jobs so that we can test that docs build without having to push them

Reply:
@seemethere What distinguishes between the two jobs? How would you trigger `build_docs` alone but always trigger the `push_docs` job on a merge or a tag?

Reply:
I gave some thoughts on this.

With the current way, `build_doc` happens on the `nightly` branch.
The `nightly` branch is managed by cronjob and the jobs are triggered by the PR https://github.com/pytorch/audio/pull/306.
With this approach, the hosted documentation will be updated once a day. I think this is fine.

But at the same time, it will be also nice if normal PR jobs include doc build process too, so that we can check how the documentation will be changed. (so the job builds the documentation html and uploads them to somewhere in CCI's artifact, which is browsable)

One solution I think of is to remove the `only` clause from the configuration YAML file (so the doc build process runs on all the PR and commits) then do the branch checking manually in the bash script.

Something like

```
if [ $CIRCLE_BRANCH = "master" ]; then
    push
fi
```

The advantage of this approach is it's easy to add release branch.

@seemethere @mattip What do you think?

Reply:
Building the docs on every PR seems like a good thing, then we could check for sphinx warnings and stop the merge until the PR fixes docs. As it stands the `build_docs` job depends on the `nightly_binary_linux_conda_py3.8`. What is the name of the python 3.8 build job on every PR that should be required instead, so the docs do not have to rebuild the package as well?

Reply:
I guess it would be `binary_linux_wheel_py3.8`

Reply:
> I guess it would be `binary_linux_wheel_py3.8`

Yes, that's correct.